### DISCUSSION TITLE
Saswell SEA801-Zigbee/SEA802-Zigbee system_mode and configuration

### DISCUSSION THREAD

**Discussion by timsterzel** [Koenkk/zigbee2mqtt#14295] [Devices]
Hello together,
I just get Hama smart home thermostats (equals to [Saswell SEA801-Zigbee/SEA802-Zigbee](https://www.zigbee2mqtt.io/devices/SEA801-Zigbee_SEA802-Zigbee.html#saswell-sea801-zigbee%252Fsea802-zigbee)) and connected them via Zigbee2MQTT with HomeAssistant.
In Zigbee2MQTT there are some settings under 'Exposes' I can make. There is the setting 'system_mode' and I'm unsure what is the correct option here. I can choose between ```off```, ```heat``` and ```auto```.
It seems like the set temperature frequently get's override by the device back to 16 degree when it is set to ```auto``` (Then also there is displayed a clock icon on the thermostat's display). If I remove this icon by clicking at the wheel of the thermostat it is not changing automatically back to 16 degrees anymore, but the devices changes to system_mode ```heat```. Is this mode the correct one or does it means it will heat all time when it is under this mode?
In the documentation I can not find an explanation of the affect of the three modes. 

At the state 'heating' it is also showing ```N/A```

**Comment by jeroen85** [comment]
I also just noticed it sometimes sets the temperature back to 16degC. But my valve (_TZE200_c88teujp) is not on auto (no clock icon), it is on heating mode.

See around 6:00
<img width="1035" alt="Screenshot 2022-10-06 at 07 15 33" src="https://user-images.githubusercontent.com/26403565/194221504-2f9198b6-9685-4cdf-8b9b-5b4b830243ec.png">

**Reply by timsterzel** [reply]
I assume the 16 degrees ist something default inside the device. Maybe it is working as fallback when it can't reach the HA central for a moment. Very strange for sure. When using the official app (depending on the branded versions), it is possible to programm heating plans which seem to get stored inside the decive directly. At least I assume that this might be the case. Maybe this kind of pre programmed heating plans might make trouble as well.

**Reply by OlafKocanda** [reply]
Do you think it is possible to reprogramm the standard value of the device? For me, with the auto mode and a temperature set it always comes back to the 16 degrees:
![grafik](https://user-images.githubusercontent.com/58265203/196224067-7ec9f2e2-26c4-4b04-8464-bc3fb05f5c9b.png)


Looks like this is a predefined value (when trusting this documentation): https://www.manualslib.com/manual/2156793/Saswell-Sea801.html?page=15#manual

**Reply by Quidamdk** [reply]
I have same thing, in auto mode it randomly sets setpoint temperature.
Did any of you find a working solution?

**Comment by jeroen85** [comment]
Maybe also related to this issue in ZHA?
https://github.com/zigpy/zha-device-handlers/issues/1815

**Reply by timsterzel** [reply]
Seems similar from behaviour, but the author there is writing "noticed and worked-around by Zigbee2MQTT developers". So usually this thing should be fixed already when I understand write.
I'm also wondering if there is a way to overwrite the build in shedule.

**Comment by joeman79** [comment]
Hello
Thank you for you post! It was driving me crazy, having the set temperature changing itself between 16 and 20.
At least I can see I am not alone and this is no misconfiguration from my part.

My setup is jeedom + zigbee2mqtt 1.28.2
My TRVs have never seen an official tuya gateway, I don't own one.
I noticed the set temperature changing by itself in a guest room. After activating value historization in jeedom, here is the schedule for every day:
from 6AM to 8AM: 20°
from 6PM to 8PM 20°
16° outside of those intervals.

There was not schedule configuration on jeedom and the TRV istself was in "auto" mode.

After following some of the links in this thread (thank you again about that), I discovered that my TRV was in scheduled mode: a little clock icon was displayed each time the display lit up.
I disabled the clock ( one press on the rotary plate) and so far, so good, it seems to work.

**Reply by tmerten** [reply]
Yes, I send `system_mode=heat` with every `current_heating_setpoint` command and this seems to be a workaround as well (I still don't know when/if that command is reset so I simply added a function that always sends both). With the original app/gateway however one could simply remove the schedule.

**Reply by joeman79** [reply]
So it is indeed thanks to your developments this behavior is possible.
For a moment, I thought that it had always been as stupid as simply disabling the schedule mode on the TRV.
Thank you!

**Reply by tmerten** [reply]
oh. It's just my personal workaround :) I am just trying to help and I am not a dev of the project (just in case this was not clear)

**Reply by avdleun** [reply]
my story with thos thermostat-knob..
i ordered 3 thermostats of the brand HI-home which is a white label of Saswell SEA802-Zigbee
My home automation consists basicly of the following components: 3 linux containers (LXC) running in a proxmox environment:
zigbee2mqtt, mosquitto broker and openHab 3.3
In the beginning they were not recognised by zigbee2mqtt.
After some digging we found that the manufactor name ('_TZE200_exfrnlow') was not mentioned in saswell.js file
After adding this name in saswell.js it worked. (we did a pull request to zigbee2.mqtt to incoperate it and this fix is accepted)
2 of the 3 thermostats worked as expected, number 3 was recognised but missed several properties.
After playing around by manually setting some missing properties #3 came also alive and was working properly.. i thought...
However I  also ran into this issue (setting itself randomly to 16 degrees)
If i set system mode to heat, the clock icon disapears and it seems to work as expected.
you can also do this manually : see https://community.home-assistant.io/t/how-to-delete-factory-set-temp-schedule-from-t0l0-tuya-sea802-z01-radiator-valve/390286
But if somebody presses the button accidently the clock icon comes up again...
(In my case by setting system mode=heat it is not possible to enable the clock manually by presssing the knob)

**Reply by joeman79** [reply]
> oh. It's just my personal workaround :) I am just trying to help and I am not a dev of the project (just in case this was not clear)

Nonetheless, my TRVs are working as expected thanks to you, it earns you some credit !

**Comment by alnospam64** [comment]
Sorry to jump in this thread with a question of my own.
By the way  I have many Saswell devices and when I set them up I always set the mode and temperature. This works for me.

One issue I have is that they announce themselves  all the time.
Is this normal?

**Reply by mwp007** [reply]
Same here. Permananently announcing itself. On Pairing I also recieved following error:

`Error collectOptions for 60a423fffeb5b8da. Error: TypeError: Cannot read property 'dp' of undefined at Object.convert (/opt/iobroker/node_modules/zigbee-herdsman-converters/lib/legacy.js:2029:32) at /opt/iobroker/node_modules/iobroker.zigbee/main.js:534:43 at StatesController.collectOptions (/opt/iobroker/node_modules/iobroker.zigbee/lib/statescontroller.js:188:17)`

**Comment by spattinson** [comment]
System_mode "auto" and "heat" are misleading mode names - "auto" means follow the schedule on the TRV which has time windows and setpoints you can set but not get. "Heat" means maintain the setpoint without any scheduling inside the TRV.
Auto would be better named "auto schedule"
Heat would be better named "auto"
